test/provider/test.pm

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+
+package test;
+
+sub rsaencdec {
+    my $prov = "OPENSSL_CONF=$ENV{IBMCA_OPENSSL_TEST_CONF} OPENSSL_MODULES=$ENV{IBMCA_TEST_PATH}";
+
+    my ($keylen, $tests, $max_file_size) = @_;
+
+    `$prov openssl list -providers | grep "name: ibmca"`;
```

Comment:
changed this to use exit(99)

The engine tests use `exit(77) unless (`$eng openssl engine -c` =~ m/ibmca/);` to skip the tests if the engine is not loaded. Not sure if this is really a good idea ? Should the tests be skipped if the engine/provider is not loaded, or is this rather an error sutuation?

Reply:
I consider this an error since `$eng` should make sure the engine is loaded.  If that is not the case, the engine failed to load and we have some error (not necessarily in the crypto operation, but in the engine itself).  I guess I have to clean up the test cases for the engine at some point.